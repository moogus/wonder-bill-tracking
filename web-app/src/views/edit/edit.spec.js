import React from 'react';
import { shallow } from 'enzyme';
import { navigate } from 'gatsby';

import EditView from './edit';

jest.mock('gatsby');

const render = ({ payment = null, updatePayment = () => {}, deletePayment = () => {} } = {}) => {
  const props = { payment, updatePayment, deletePayment };
  return shallow(<EditView {...props} />);
};

describe('EditView', () => {
  describe('payment is not passed to the Edit route', () => {
    let editView;
    beforeAll(() => {
      jest.clearAllMocks();
      editView = render();
    });

    test('default render is as expected', () => expect(editView).toMatchSnapshot());

    test('navigate is called with "/"', () => expect(navigate).toHaveBeenCalledWith('/'));
  });

  describe('payment is passed to the Edit route', () => {
    let editView;
    const payment = {
      id: 0,
      name: 'foo',
      amount: 'bar',
      startDate: 'baz',
      frequency: 'always',
    };

    beforeAll(() => {
      jest.clearAllMocks();
      editView = render({ payment });
    });

    test('default render is as expected', () => expect(editView).toMatchSnapshot());

    test('navigate is called with "/"', () => expect(navigate).not.toHaveBeenCalled());
  });
});
