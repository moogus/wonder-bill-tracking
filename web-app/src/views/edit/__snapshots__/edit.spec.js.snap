// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditView payment is not passed to the Edit route default render is as expected 1`] = `
<PaymentForm
  buttonMessage="Save"
  description="If you'd like to edit your bill you can change the details below"
  onClick={[Function]}
  onDelete={[Function]}
  payment={null}
  title=""
/>
`;

exports[`EditView payment is passed to the Edit route default render is as expected 1`] = `
<PaymentForm
  buttonMessage="Save"
  description="If you'd like to edit your bill you can change the details below"
  onClick={[Function]}
  onDelete={[Function]}
  payment={
    Object {
      "amount": "bar",
      "frequency": "always",
      "id": 0,
      "name": "foo",
      "startDate": "baz",
    }
  }
  title="foo"
/>
`;
